yammbs/analysis.py

```diff
@@ -152,35 +150,110 @@ def get_internal_coordinate_rmsds(
         _to_geometric_molecule(molecule=molecule, coordinates=target),
     )
 
-    types: dict[str, type | None] = {
-        _type: {
-            "Bond": Distance,
-            "Angle": Angle,
-            "Dihedral": Dihedral,
-            "Improper": OutOfPlane,
-        }.get(_type)
-        for _type in _types
+    _mapping = {
+        "Bond": Distance,
+        "Angle": Angle,
+        "Dihedral": Dihedral,
+        "Improper": OutOfPlane,
     }
+    types: dict[str, type] = {_type: _mapping[_type] for _type in _types}
+
+    internal_coordinates: dict[str, dict[tuple[int, ...], tuple[int, int]]] = dict()
 
-    internal_coordinates = {
-        label: [
-            (
-                internal_coordinate.value(reference),
-                internal_coordinate.value(target),
+    for label, internal_coordinate_class in types.items():
+        internal_coordinates[label] = dict()
+
+        for internal_coordinate in _generator.Internals:
+            if not isinstance(internal_coordinate, internal_coordinate_class):
+                continue
+
+            if isinstance(internal_coordinate, Distance):
+                key = tuple(
+                    (
+                        internal_coordinate.a,
+                        internal_coordinate.b,
+                    ),
+                )
+
+            if isinstance(internal_coordinate, Angle):
+                key = tuple(
+                    (
+                        internal_coordinate.a,
+                        internal_coordinate.b,
+                        internal_coordinate.c,
+                    ),
+                )
+
+            if isinstance(internal_coordinate, (Dihedral, OutOfPlane)):
+                key = tuple(
+                    (
+                        internal_coordinate.a,
+                        internal_coordinate.b,
+                        internal_coordinate.c,
+                        internal_coordinate.d,
+                    ),
+                )
```

Comment:
Quick question: does this key follow the same as a SMIRNOFF improper -- i.e. is b the central atom?

Reply:
Oh great catch - was just copying code, will look at & add tests